api/nodes/views.py

```diff
@@ -109,8 +109,13 @@ def perform_destroy(self, instance):
         user = self.request.user
         auth = Auth(user)
         node = self.get_object()
-        node.remove_node(auth=auth)
-        node.save()
```

Comment:
Does the node model not raise an exception here? I presume that's what's causing the 500. Ideally, we should just catch the exception and return the 403 if it happens.